opencontainers/struct.py

```diff
@@ -293,15 +303,43 @@ def add(self, name, value):
         """
         Add a value to an existing attribute, normally when used by a client
         """
+        if value is None:
+            return
         if name not in self.attrs:
             bot.exit("%s is not a valid attribute." % name)
         attr = self.attrs[name]
 
-        # If we already have a value, update it
-        if attr.value and isinstance(attr.value, dict):
-            value.update(attr.value)
-        elif attr.value and isinstance(attr.value, list):
-            value = list(set(attr.value + [value]))
+        valueType = type(value)
+        if attr._is_struct():
+            # Transform value to Struct if not already
+            if not is_struct(valueType):
+                value = attr.attType().load(value)
+            if attr.attType in [StrStruct, IntStruct]:
```

Comment:
Adding an int/string to another? is this the behavior we want?

Reply:
I don't know.
In another comment (https://github.com/vsoch/oci-python/pull/14#discussion_r671389313) you noticed that replacing the value might not be the thing we want.
So maybe we should only allow dict and list as type for value?!

Reply:
My thinking is that add should explicitly be for things you can add/append to (like list or dict). If someone wants to change a value entirely they can just use set. I can see the argument for wanting add for an int (to add a number to an existing one) but I'd say we should wait until someone actually speaks up about that use case.

Reply:
But imho we should have a `Struct.set()` then.
Because for now one have to use `Struct.attrs["name"].set("value")` to change the value of an attribute.
But you asked me to revert the commit which added `Struct.set()`
So I don't understand what we're up to..

Reply:
I thought we already have .set() on lines 75/84?

Reply:
Indeed but only in `StructAttr` not in `Struct`.
So `Struct.attrs["name"].set("value")` is the only way to change a value of an attribute.
Don't know if this the way one should set a value when we have a `Struct.add("name", "value")` to append to a list.

Reply:
Oh I see. I think Struct.add with name and value would make sense to append to a list, do you think not? Let's discuss what you think would be most intuitive!